### PR TITLE
Allow preserving the initial bone pose in rest fixer

### PR DESCRIPTION
**Pull request by lyuma**
Adds a "Preserve Initial Pose" boolean option to the Skeleton3D Rest Fixer settings `retarget/rest_fixer/preserve_initial_pose`
This option preserves the bone pose as it would have been before silhouette adjustment.

Basically, ufbx calculates the rest pose differently in some models which use skin binds. Rest pose should only used in the retargeting process, but prior to this change, the retargeting flow left the model in rest pose and discarded the actual bone_pose, which now makes scene inheritance dependent on the retargeter's rest pose.

Before:
before_preserve_initial_pose
Show Rest Only (same before and after):
tpose_rest
After:
preserve_initial_pose

### CODE REVIEW

**Comment by fire** [comment]
Is there a reason why this cannot be defaulted on?

**Comment by TokageItLab** [comment]
Can you explain what the actual case this solves is?
Do you mean that if we change the pose after importing the Skeleton and then re-import it, we will have a problem?

**Comment by lyuma** [comment]
A model may be in some pose upon export from the DCC. ufbx has some code to guess the rest pose from skin binds, and the retarget silhouette fixer will also attempt to calculate a new rest pose.

The retarget system operates entirely on bone rest. however, some code at the end of the retarget flow copies the bone rest to the bone pose, losing the initial pose (which may have been important).

It is probably not important for some usecases, but I need the initial pose in Unidot because scene inheritance may depend on knowing the initial pose.
@fire it should not default on for two reasons:
First, many users probably prefer to see their model in t-pose. Second, it would break compatibility of scene inheritance if we changed the default. The behavior prior to this change is to discard the initial pose and replace with t-pose

**Comment by lyuma** [comment]
From what I understand, your concern is that a model is imported which perhaps only has a subset of animation tracks (for example, Left and Right arms)

> Is it only the pose that appears after import that is affected?

Correct, this is the only thing affected

> Is the animation also affected?

It should not be affected by the initial pose. My understanding is the rest fixer guarantees that animation keys are added for all bones in the bone map (unless the user manually disables this)

Basically, the animation keys would be relative to the initial pose. However, during retarget we may adjust the model's pose and it is critical that the animation continue to look correct. Therefore, Godot must already be adding the necessary keys, and animations should play regardless of pose.

My assumption here is that the animation optimizer won't remove tracks that are equal to the bone pose: if so, this would cause t-pose animations to have no tracks in current version of godot.

> Can animations be shared correctly between multiple models using this workflow?

I do not see how the initial bone pose would affect animation sharing. The initial bone pose should be lost when animations are played.

This PR is really a matter of "correctness" or a "technicality". If a model is initially posed as doing a handshake without retarget, it should also be initially posed in a handshake with retarget. Retarget should be adjusting the rolls of the bones, but not affect the visual result of the model import.
The goal of this PR is to allow the user to preserve the imported result visually, while performing retargeting correctly..

To be perfectly honest, this PR is not critical. But it feels like the right thing to do, to avoid losing information.

**Comment by TokageItLab** [comment]
If I understand correctly, we could change the option name to like `reset all bone poses after import` with inverting the boolean.

**Comment by lyuma** [comment]
Reset all Bone Poses after Import sounds good!

As for why it's in retargeter, it is because that is the only place I could guarantee correctness.
Outside retargeter, the bone rest will usually match bone pose anyway. The ufbx behavior is new and only affects bone rest.

However, if we blindly reset bone poses outside of rest fixer, then we may need to add additional animation tracks for each bone we changed which means there will be possible side effects of using this option that the user may not expect.

To contrast, in retargeting, it is expected that all animation tracks exist after retargeting is done, so there will be no side effects of adjusting the bone pose.

To summarize:
1. When silhouette fixer (or #88824) is not enabled, he bone pose is almost always equal to the bone rest. The case where this option has an effect is rare.
2. When rest fixer is not enabled, it is not clear that we can safely adjust the initial bone pose without disrupting animations.

**Comment by TokageItLab** [comment]
After renaming it to `Reset All Bone Poses After Import`, I think the default should be False if you place it outside the retarget or True if you place it inside the retarget. For compatibility, it may be fine to place it in the retarget with the default be True.

File: editor/import/3d/post_import_plugin_skeleton_rest_fixer.cpp [hunk 1]
```diff
@@ -603,6 +609,30 @@ void PostImportPluginSkeletonRestFixer::internal_process(InternalImportCategory
 					}
 				}
 			}
+			if (p_options.has("retarget/rest_fixer/reset_all_bone_poses_after_import") && !bool(p_options["retarget/rest_fixer/reset_all_bone_poses_after_import"])) {
```
Comment (TokageItLab) [inline editor/import/3d/post_import_plugin_skeleton_rest_fixer.cpp:612]: If store the bool results of this earlier, you can avoid unnecessary calculations for some pose changes in line 117-120 and 131 as well.

**Review by TokageItLab — APPROVED** [review]
It makes sense as an optional feature. As far as I can look in the code, rest has not been changed, so I assume this change is reasonably safe.

File: editor/import/3d/post_import_plugin_skeleton_rest_fixer.cpp (hunk 1 above)
Comment (lyuma) [reply editor/import/3d/post_import_plugin_skeleton_rest_fixer.cpp:612]: I considered this, but the number of calculations are low, and doing them in that part of the code preserves correctness (counteracting scale changes).

**Review by fire — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
